### DISCUSSION TITLE
How to get references to elements

### DISCUSSION THREAD

**Discussion by DusterTheFirst** [DioxusLabs/dioxus#383] [Q&A]
I wish to call functions on the elements that have been layed out by dioxus, but doing so requires I obtain a `HTMLSomethingElement` type from `web_sys`. In react this would be done with refs (https://reactjs.org/docs/refs-and-the-dom.html). Is there any way to achieve this functionality with dioxus?

Another side question, is it possible to write event handlers that execute javascript or have access to the target's dom element?

**Comment by jkelleyrtp** [comment]
> I wish to call functions on the elements that have been layed out by dioxus, but doing so requires I obtain a `HTMLSomethingElement` type from `web_sys`. In react this would be done with refs (https://reactjs.org/docs/refs-and-the-dom.html). Is there any way to achieve this functionality with dioxus?
> 
> Another side question, is it possible to write event handlers that execute javascript or have access to the target's dom element?

We haven't implemented domrefs yet since we need to support so many targets - not just wasm. For wasm, I've been suggesting using use_effect/use_future to get an element by its ID. I wish we had an example but none of our primary examples are set to build on wasm yet.

```rust
use_future(&cx, || async move {
    let el = getElementById("some id");
});

```
